src/main/io/gps.c

```diff
@@ -357,122 +425,128 @@ void gpsInit(void)
         return;
     }
 
-    while (gpsInitData[gpsData.baudrateIndex].baudrateIndex != gpsPortConfig->gps_baudrateIndex) {
-        gpsData.baudrateIndex++;
-        if (gpsData.baudrateIndex >= GPS_INIT_DATA_ENTRY_COUNT) {
-            gpsData.baudrateIndex = DEFAULT_BAUD_RATE_INDEX;
+    // set the user's intended baud rate
+    gpsData.userBaudRateIndex = 0;
```

Comment:
duplicate from line 400

Reply:
Thanks.
I very much want to re-name the baud rate initialisation enum variables to something sensible.
For example, `gpsInitData_t` and `static const gpsInitData_t gpsInitData[] = {`, are specific to baud rate assignment, but their names do not convey their purpose well, not at all.